### PR TITLE
Post management

### PR DESCRIPTION
**Pull request by akash-1318**
# Added post features like - upload post, edit post, delete post.
## Files to check -
1. src/features/postSlice.js
2. src/services/postServices.js
3. src/components/user-profile/user-profile.js
4. src/components/other-profiles/other-profiles.js
5. src/components/new-post/new-post.js
6. src/components/edit-post-modal/edit-post-modal.js

### CODE REVIEW

File: src/backend/db/posts.js [hunk 1]
```diff
@@ -10,9 +10,10 @@ export const posts = [
   {
     _id: uuid(),
     content:
-      "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.",
+      "Hard work beats talent when talent doesn't work hard.",
+    pic : '',
     likes: {
-      likeCount: 0,
+      likeCount: 2,
       likedBy: [],
```
Comment (znjs) [inline src/backend/db/posts.js:17]: if likedCount is 2 
you will need to fill the data in likedBy array too

File: src/components/edit-post-modal/edit-post-modal.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    let file = e.target.files[0];
+
+    const getImgUrl = (file) => {
+      return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = (error) => reject(error);
+      });
+    };
```
Comment (znjs) [inline src/components/edit-post-modal/edit-post-modal.js:40]: avoid nested functions
use try-catch block for error handling

File: src/features/postSlice.js [hunk 3]
```diff
@@ -15,6 +16,56 @@ export const getUserPostsData = createAsyncThunk("post/getUserPostsData", async(
     }
 })
 
+export const getAllPostsData = createAsyncThunk("post/getAllPostsData", async() => {
+    try{
+        const resp = await getAllPosts()
+        return resp.data
+    } catch(error){
+        return rejectWithValue(error)
```
Comment (znjs) [inline src/features/postSlice.js]: rejectWithValue doesn't work directly without thunkAPI

**Review by znjs — APPROVED** [review]
- good naming conventions used
- clean folder structure

**Review by srv-pranjal — APPROVED** [review]
Changes work as expected.
Structured code.
Clean UI

**Pull request merged**